src/app/Library/Validation/Rules/BackpackCustomRule.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\Validation\Rules;
+
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade;
+use Closure;
+use Illuminate\Contracts\Validation\DataAwareRule;
+use Illuminate\Contracts\Validation\Rule;
+use Illuminate\Contracts\Validation\ValidationRule;
+use Illuminate\Contracts\Validation\ValidatorAwareRule;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Validator;
+
+/**
+ * @method static static itemRules()
+ */
+abstract class BackpackCustomRule implements ValidationRule, DataAwareRule, ValidatorAwareRule
+{
+    /**
+     * @var \Illuminate\Contracts\Validation\Validator
+     */
+    protected $validator;
+
+    protected array $data;
+
+    public array $fieldRules = [];
+
+    public ?Model $entry;
+
+    public bool $implicit = true;
+
+    public function __construct()
+    {
+        $entry = CrudPanelFacade::getCurrentEntry();
+        $this->entry = $entry !== false ? $entry : null;
+    }
+
+    public static function field(string|array|ValidationRule|Rule $rules = []): self
+    {
+        $instance = new static();
+        $instance->fieldRules = self::prepareRules($rules);
+
+        return $instance;
+    }
+
+    /**
+     * Run the validation rule.
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @param  Closure(string): \Illuminate\Translation\PotentiallyTranslatedString  $fail
+     * @return void
+     */
+    public function validate(string $attribute, mixed $value, Closure $fail): void
+    {
+        // is the extending class reponsability the implementation of the validation logic
+    }
+
+    /**
+     * Set the performing validator.
+     *
+     * @param  \Illuminate\Contracts\Validation\Validator  $validator
+     * @return $this
+     */
+    public function setValidator($validator)
+    {
+        $this->validator = $validator;
+
+        return $this;
+    }
+
+    /**
+     * Set the data under validation.
+     *
+     * @param  array  $data
+     * @return $this
+     */
+    public function setData($data)
+    {
+        $this->data = $data;
+
+        return $this;
+    }
+
+    public function getFieldRules(): array
+    {
+        return tap($this->fieldRules, function ($rule) {
+            if (is_a($rule, BackpackCustomRule::class, true)) {
+                $rule = $rule->getFieldRules();
+            }
+
+            return $rule;
+        });
+    }
+
+    public function validateFieldRules(string $attribute, mixed $value, Closure $fail): void
+    {
+        $validator = Validator::make([$attribute => $value], [
+            $attribute => $this->getFieldRules(),
+        ], $this->validator->customMessages, $this->validator->customAttributes);
+
+        if ($validator->fails()) {
+            foreach ($validator->errors()->messages()[$attribute] as $message) {
+                $fail($message)->translate();
+            }
+        }
+    }
+
+    protected static function prepareRules($rules)
```

Comment:
Isn't this actually `getRulesArray()`? That's what it does, right? Whatever the rules are, it returns an array of rules. Right? This stood out to me because with a name like `prepareRules()` I'd expect it to be a procedure (return nothing) not a function (return something).

```suggestion
    protected static function getRulesArray($rules)
```

Reply:
Alternatively, if we don't want to confuse this with a GETTER (since it's a static method), we could call it `convertRulesToArray()` or `turnRulesIntoArray()` something like that. Right?